articles/153_roslaunch_remote_launch.md

```diff
@@ -0,0 +1,185 @@
+| get\_asyncio\_future | Optional[asyncio.Future] | Override. Return an asyncio Future, used to let the launch system know when we're done. |
+
+
+#### launch.actions.ExecuteRemoteSSH
+Inherits from `launch.actions.ExecuteRemote` and manages the ssh connection
+##### Members
+Inherited
+##### Methods
+| Name | Type | Description |
+|---|---|---|
+| open\_ssh\_connection | SSHConnection | Object for interacting with the SSH connection |
+| exec\_remote\_command | void | Executes a command on the remote machine
+
+#### launch.substitutions.RemoteSubstitution
+Uses a connection to a machine to run a command remotely and retuns the result as a string
+##### Members
+##### Constructor
+| Argument | Type | Description |
+|---|---|---|
+| command | Text | The command to run on the remote machine |
+| machine | `launch.descriptions.Machine` | An object decribing the machine that the command should be run on. |
+##### Methods
+| Name | Type | Description |
+|---|---|---|
+| describe | Text | Override method. Returns a description of this substitution as a string. |
+| perform  | Text | Override method. Perform the substitution, given the launch context, and return it as a string. |
+
+
+## ROS
+
+### Nodes
+#### launch\_ros.LaunchServiceNode
+This node runs on each machine in a system is is repsonsible for launching and managing remote processes.
+#### Params
+| Name | Type | Description |
+|---|---|---|
+| allowable\_processes | List[Executable] | List of executables the node is allowed to launch |
+##### Topics
+###### Sent
+- heartbeat
+###### Received
+##### Services
+- query\_status
+- start\_process
+- stop\_process
+- shutdown
+
+### Messages
+#### Heartbeat
+Contains basic status information and enough information for launch related tools to find the node.
+DDS might already provide enough info making this superfluous.
+##### Fields
+| Name | Type | Description |
+|---|---|---|
+| timestamp  | `builtin_interface.Timestamp` | Time at which heartbeat message was sent |
+| hostname   | String | Hostname of machine sending the status |
+| num\_procs | int | Number of processes currently being tracked by the LaunchServiceNode. |
+
+#### ProcessStatus
+##### Fields
+| Name | Type | Description |
+|---|---|---|
+| timestamp       | `builtin_interface.Timestamp` | Time at which message was sent |
+| process\_name   | String | Name of the process |
+| process\_id     | int | A unique numerical identifier for the process. This value is generated by the launch system and should not be expected to be the same as the proc id on the machine. |
+| process\_status | int | An enumerated status value. STOPPED=0; RUNNING=1; SUSPENDED=2; |
+| mem\_usage      | double | Percent of memory used by the process |
+| processor\_load | double | Percent of processor load coming from this process |
+| err\_msg        | String | Error string |
+
+### Services
```

Comment:
Noting that each LaunchServiceNode will need to namespace itself to make service calls uniquely-fulfillable